### DISCUSSION TITLE
Raspberry Pi Zero agent -  Illegal Instruction

### DISCUSSION THREAD

**Discussion by stevenbowyer** [wazuh/wazuh#18380] [General]
Does the agent support the raspberry pi zero?

On same image on a Pi4 it installs fine. But on a zero I get "illegal instruction".

Difference being the arm version and instruction set. Zero is ARMv6 and Pi4 is ARMv7. 

Is there a way to install agent on a zero?

**Comment by cjsaraiva** [comment]
Hi, same problem here:

 description: ARMv6-compatible processor rev 7 (v6l)
    product: Raspberry Pi Model B Rev 1
    width: 32 bits

systemctl start wazuh-agent

 Illegal instruction
Jan 04 15:00:08 xxxxx: wazuh-execd: Configuration error. Exiting
Jan 04 15:00:08 xxxxx systemd[1]: wazuh-agent.service: Control process exited,

Installer: wazuh-agent_4.7.1-1_armhf.deb

**Comment by tug-benson** [comment]
Hello, 

Architecture:           armv6l
product: Raspberry Pi Model B Rev 1
Model name:           ARM1176
width: 32 bits

:~# /var/ossec/bin/wazuh-execd -d
Illegal instruction

Agent seems hasn't to be non compile it on model b

Installer: wazuh-agent_4.12.0-1_armhf.deb

**Comment by DaijoubuKun** [comment]
I've been working on getting a build going for the first gen PIs as well, and even when I build from source, I can't seem to get it to work either. I get the same error about an Illegal Instruction. Running the wazuh-agentd program with strace revels there was an Illegal Opcode, which means something in there is calling a CPU instruction that the CPU doesn't know what to do with it. I'm not sure if somewhere along the line I'm building the agent using the wrong "arm" arch or what. 

I'm still going to crack away at this, but I'm getting to the point where it really isn't worth it because I'm spending so much time trying just to get the agent to build at all.

**Comment by cjsaraiva** [comment]
Hi,

I just choose to change to Raspberry PI 2.

CS

On Tue, 7 Oct 2025 at 21:42, DaijoubuKun ***@***.***> wrote:

> I've been working on getting a build going for the first gen PIs as well,
> and even when I build from source, I can't seem to get it to work either. I
> get the same error about an Illegal Instruction. Running the wazuh-agentd
> program with strace revels there was an Illegal Opcode, which means
> something in there is calling a CPU instruction that the CPU doesn't know
> what to do with it. I'm not sure if somewhere along the line I'm building
> the agent using the wrong "arm" arch or what.
>
> I'm still going to crack away at this, but I'm getting to the point where
> it really isn't worth it because I'm spending so much time trying just to
> get the agent to build at all.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wazuh/wazuh/discussions/18380#discussioncomment-14619399>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AX3K6262DURP5AO7EMM537T3WQQTTAVCNFSM6AAAAACAJITLLCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINRRHEZTSOI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Cumps

CS

**Comment by DaijoubuKun** [comment]
I feel like I got really close to figuring this out. I did a big 'ol rewrite of some CMake files in hopes of solving this, but now I've run into one major issue. Right at the end of the build, I get these lines:

```
    LINK libwazuh.a
ar: `u' modifier ignored since `D' is the default (see `U')src/external# 
    RANLIB libwazuh.a
    CC wazuh-agentd
/usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_load_8'
/usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_fetch_or_8'
/usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_is_lock_free'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:2112: wazuh-agentd] Error 1
make[1]: Leaving directory '/root/wazuh/wazuh-4.13.0/src'
make: *** [Makefile:865: agent] Error 2
```

I just can't seem to get past this last part with libwazuhext.so throwing errors with ld. I don't know if this can be ignored somehow or if I need to figure out why libwazuhext.so isn't building correctly. I'm just not sure. Any help would be appreciated.

**Comment by cjsaraiva** [comment]
Hi,

I also follow that path and give it up....

CS

On Mon, 27 Oct 2025 at 22:59, DaijoubuKun ***@***.***> wrote:

> I feel like I got really close to figuring this out. I did a big 'ol
> rewrite of some CMake files in hopes of solving this, but now I've run into
> one major issue. Right at the end of the build, I get these lines:
>
>     LINK libwazuh.a
> ar: `u' modifier ignored since `D' is the default (see `U')src/external#
>     RANLIB libwazuh.a
>     CC wazuh-agentd
> /usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_load_8'
> /usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_fetch_or_8'
> /usr/bin/ld: libwazuhext.so: undefined reference to `__atomic_is_lock_free'
> collect2: error: ld returned 1 exit status
> make[1]: *** [Makefile:2112: wazuh-agentd] Error 1
> make[1]: Leaving directory '/root/wazuh/wazuh-4.13.0/src'
> make: *** [Makefile:865: agent] Error 2
>
> I just can't seem to get past this last part with libwazuhext.so throwing
> errors with ld. I don't know if this can be ignored somehow or if I need to
> figure out why libwazuhext.so isn't building correctly. I'm just not sure.
> Any help would be appreciated.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wazuh/wazuh/discussions/18380#discussioncomment-14798491>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AX3K6246P5P7ZWJUOTPY36L3Z2PVFAVCNFSM6AAAAACAJITLLCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINZZHA2DSMI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Cumps

CS
